File: uploads/date.js
import express from 'express';
import bodyParser from 'body-parser';
import admin from 'firebase-admin';
import path from 'path';
import cors from 'cors';
import dotenv from 'dotenv';

const app = express();
dotenv.config();

// Initialize Firebase Admin
app.use(cors());
const serviceAccount = path.resolve(process.env.APP_FIREBASE_CREDENTIALS);
admin.initializeApp({
  credential: admin.credential.cert(serviceAccount),
});

// Get a reference to the Firestore database
const db = admin.firestore();

// Middleware to parse JSON
app.use(bodyParser.json());

app.post('/api/date', async (req, res) => {
  const { date } = req.body;

  if (date) {
    try {
      // Add the date to the 'dataTambahan' collection in the 'date' document
      const docRef = db.collection('dataTambahan').doc('date');
      await docRef.set({ date: date });

      res.status(200).json({ message: 'Date successfully added to Firestore', date });
    } catch (error) {
      console.error('Error adding date to Firestore:', error);
      res.status(500).json({ message: 'Failed to save date', error });
    }
  } else {
    res.status(400).json({ message: 'No date provided' });
  }
});

const PORT = 3000;
app.listen(PORT, () => console.log(`Server running on port ${PORT}`));

File: uploads/date.py
import os
import json
import firebase_admin
from firebase_admin import credentials, firestore
from flask import Flask, request, jsonify
from dotenv import load_dotenv
from flask_cors import CORS

# Load environment variables
load_dotenv()

app = Flask(__name__)
CORS(app)

# Initialize Firebase Admin
cred = credentials.Certificate(os.path.abspath(os.getenv("APP_FIREBASE_CREDENTIALS")))
firebase_admin.initialize_app(cred)

db = firestore.client()

@app.route('/api/date', methods=['POST'])
def add_date():
    data = request.get_json()
    date = data.get('date')

    if date:
        try:
            # Add the date to the 'dataTambahan' collection in the 'date' document
            doc_ref = db.collection('dataTambahan').document('date')
            doc_ref.set({'date': date})

            return jsonify({"message": "Date successfully added to Firestore", "date": date}), 200
        except Exception as e:
            print("Error adding date to Firestore:", e)
            return jsonify({"message": "Failed to save date", "error": str(e)}), 500
    else:
        return jsonify({"message": "No date provided"}), 400
    
# @app.route('/api/date', methods=['GET'])
# def get_date():
#     try:
#         # Get the date from the 'dataTambahan' collection in the 'date' document
#         doc_ref = db.collection('dataTambahan').document('date')
#         doc = doc_ref.get()
#         if doc.exists:
#             return jsonify({"date": doc.to_dict().get("date")}), 200
#         else:
#             return jsonify({"message": "No date found"}), 404
#     except Exception as e:
#         print("Error retrieving date from Firestore:", e)
#         return jsonify({"message": "Failed to retrieve date", "error": str(e)}), 500


if __name__ == '__main__':
    PORT = 5001
    app.run(debug=True, port=PORT)